lib/autocomplete-manager.coffee

```diff
@@ -272,18 +186,166 @@ class AutocompleteManager
     @editor.insertText(match.word)
     @editor.setSelectedBufferRanges(newSelectedBufferRanges)
 
-  # Private: Sets the current buffer, starts listening to change events and delegates
-  # them to #onChanged()
+  # Public: Finds and returns the content before the current cursor position
+  #
+  # selection - The {Selection} for the current cursor position
+  #
+  # Returns {String} with the prefix of the {Selection}
+  prefixOfSelection: (selection) ->
+    selectionRange = selection.getBufferRange()
+    lineRange = [[selectionRange.start.row, 0], [selectionRange.end.row, @editor.lineTextForBufferRow(selectionRange.end.row).length]]
+    prefix = ''
+    wordRegex = /\b\w*[a-zA-Z_-]+\w*\b/g
```

Comment:
Right now the only time I see this is in `atom-typescript-tools` since the typescript autocomplete providers provides the completions with the `$` in them.